### ISSUE TITLE
Periodic schedule is pushed back indefinitely.

### ISSUE THREAD

**Issue by NG1973** [borgbase/vorta#1291]
I have Vorta configured to backup every 4 hours. However, since updating to version 0.8.4 the scheduling is failing to run and the log shows the time for the next backup being incremented without a backup occuring. Excerpt from the log below:

```
2022-05-10 12:49:33,415 - vorta.scheduler - DEBUG - Scheduling next run for 2022-05-10 16:49:33.415096
2022-05-10 13:04:33,154 - vorta.scheduler - DEBUG - Refreshing all scheduler timers
2022-05-10 13:04:33,157 - vorta.scheduler - INFO - Setting timer for profile 2
2022-05-10 13:04:33,158 - vorta.scheduler - DEBUG - Scheduling next run for 2022-05-10 17:04:33.158586
2022-05-10 13:19:33,157 - vorta.scheduler - DEBUG - Refreshing all scheduler timers
2022-05-10 13:19:33,160 - vorta.scheduler - INFO - Setting timer for profile 2
2022-05-10 13:19:33,161 - vorta.scheduler - DEBUG - Scheduling next run for 2022-05-10 17:19:33.161549
2022-05-10 13:34:33,254 - vorta.scheduler - DEBUG - Refreshing all scheduler timers
2022-05-10 13:34:33,257 - vorta.scheduler - INFO - Setting timer for profile 2
```

**Environment**
- OS: Manjaro
- Vorta version: 0.8.4
- Installed from: AUR

**Comment by NG1973** [comment]
Apologies for the late reply. Scheduled backups are now running and the 'Next Backup:' time is advanced 4 hours as it should. I'm not sure what changed, I had restarted the machine a few times since updating.

The only minor thing is that the backups used to be scheduled to run on the hour, now each backup is scheduled for 4 minutes past the hour, it's only minor and as the backup occurs I'm not too bothered. This can probably be closed now.

**Comment by real-yfprojects** [comment]
> Apologies for the late reply. Scheduled backups are now running and the 'Next Backup:' time is advanced 4 hours as it should. I'm not sure what changed, I had restarted the machine a few times since updating.
> 

Did you start a manual backup?


> The only minor thing is that the backups used to be scheduled to run on the hour, now each backup is scheduled for 4 minutes past the hour, it's only minor and as the backup occurs I'm not too bothered. 

Some kind of rounding could be implemented, currently it actually schedules every 4 hours without rounding the hour.

> This can probably be closed now.

No, there is a code issue to this. I will try to fix it by the end of this week.

**Comment by NG1973** [comment]
> Did you start a manual backup?
> 
> >

No, an update was triggered by a Pacman Hook but this does not involve Vorta. I did refresh the archive list though.



> Some kind of rounding could be implemented, currently it actually schedules every 4 hours without rounding the hour.
> 
> >

There does appear to be some time creep in the scheduling (note the seconds), these are the relevant bits from my log:
`2022-05-12 11:50:05,240 - vorta.scheduler - DEBUG - Scheduling next run for 2022-05-12 12:04:49.899452`
`2022-05-12 12:04:54,226 - vorta.scheduler - DEBUG - Scheduling next run for 2022-05-12 16:04:53.412570`
`2022-05-12 16:05:04,213 - vorta.scheduler - DEBUG - Scheduling next run for 2022-05-12 20:04:57.324970`



> No, there is a code issue to this. I will try to fix it by the end of this week.

Thank you. I only suggested closing the issue becuase I thought the issue may be specific to me.
